tests/app/questionnaire/routing/test_when_rule_evaluator.py

```diff
@@ -0,0 +1,965 @@
+from app.questionnaire.relationship_location import RelationshipLocation
+from app.questionnaire.routing.operator import Operator
+from app.questionnaire.routing.when_rule_evaluator import WhenRuleEvaluator
+from tests.app.questionnaire.test_value_source_resolver import get_list_items
+
+now = datetime.utcnow()
+now_as_yyyy_mm_dd = now.strftime("%Y-%m-%d")
+
+
+def get_mock_schema():
+    schema = Mock(
+        QuestionnaireSchema(
+            {
+                "questionnaire_flow": {
+                    "type": "Linear",
+                    "options": {"summary": {"collapsible": False}},
+                }
+            }
+        )
+    )
+    return schema
+
+
+def get_when_rule_evaluator(
+    rule: dict,
+    schema: QuestionnaireSchema = None,
+    answer_store: AnswerStore = AnswerStore(),
+    list_store: ListStore = ListStore(),
+    metadata: Optional[dict] = None,
+    location: Union[Location, RelationshipLocation] = Location(
+        section_id="test-section", block_id="test-block"
+    ),
+    routing_path_block_ids: Optional[list] = None,
+):
+    if not schema:
+        schema = get_mock_schema()
+        schema.is_repeating_answer = Mock(return_value=True)
+        schema.get_default_answer = Mock(return_value=None)
+
+    return WhenRuleEvaluator(
+        rule=rule,
+        schema=schema,
+        metadata=metadata or {},
+        answer_store=answer_store,
+        list_store=list_store,
+        location=location,
+        routing_path_block_ids=routing_path_block_ids,
+    )
+
+
+@pytest.mark.parametrize(
+    "rule, expected_result",
+    [
+        ({Operator.NOT: [False]}, True),
+        ({Operator.AND: [True, True]}, True),
+        ({Operator.OR: [True, False]}, True),
+        ({Operator.EQUAL: ["Yes", "Yes"]}, True),
+        ({Operator.NOT_EQUAL: ["Yes", "No"]}, True),
+        ({Operator.GREATER_THAN: [3, 1]}, True),
+        ({Operator.GREATER_THAN_OR_EQUAL: [1, 1]}, True),
+        ({Operator.LESS_THAN: [0.5, 1]}, True),
+        ({Operator.LESS_THAN_OR_EQUAL: [0.5, 0.5]}, True),
+        ({Operator.IN: ["Yes", ["Yes"]]}, True),
+        ({Operator.ANY_IN: [["Yes"], ["Yes", "No"]]}, True),
+        ({Operator.ALL_IN: [["Yes", "No"], ["Yes", "No"]]}, True),
+        # Test inverse
+        ({Operator.NOT: [True]}, False),
+        ({Operator.AND: [True, False]}, False),
+        ({Operator.OR: [False, False]}, False),
+        ({Operator.EQUAL: ["Yes", "No"]}, False),
+        ({Operator.NOT_EQUAL: ["Yes", "Yes"]}, False),
+        ({Operator.GREATER_THAN: [1, 3]}, False),
+        ({Operator.GREATER_THAN_OR_EQUAL: [0.5, 1]}, False),
+        ({Operator.LESS_THAN: [1, 0.5]}, False),
+        ({Operator.LESS_THAN_OR_EQUAL: [1, 0.5]}, False),
+        ({Operator.IN: ["No", ["Yes"]]}, False),
+        ({Operator.ANY_IN: [["Maybe"], ["Yes", "No"]]}, False),
+        ({Operator.ALL_IN: [["Yes", "Maybe"], ["Yes", "No"]]}, False),
+    ],
+)
+def test_all_boolean_operators_as_rule(rule, expected_result):
```

Comment:
Probably don't need the all, (the other tests don't have this), specially as all does mean something else in this context (i.e all_in)